### PR TITLE
363:Force update parent version with real version

### PR DESCRIPTION
**Pull request by akilantech**
If parent version is RELEASE or LATEST, update-parent does not update it to real version. hence, adding this feature to replace RELEASE or LATEST with real version.

### CODE REVIEW

**Comment by khmarbaise** [comment]
Could you rebase against current master....

**Comment by akilantech** [comment]
start working on this. missed your past message

**Comment by bmarwell** [comment]
Will take a look next week, thanks.

**Comment by akilantech** [comment]
Any luck @bmarwell  ?

File: src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java [hunk 1]
```diff
@@ -54,6 +54,14 @@
     @Parameter( property = "parentVersion", defaultValue = "null" )
     protected String parentVersion = null;
 
+    /**
+     * to update parent version by force when it is RELEASE or LATEST
+     *
+     * @since 2.8
+     */
+    @Parameter( property = "forceUpdate", defaultValue = "False" )
+    protected Boolean forceUpdate = null;
```
Comment (olamy) [inline src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java:63]: maybe should be `boolean` rather than `Boolean` to avoid NPE.

**Review by olamy — CHANGES_REQUESTED** [review]

File: src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java (hunk 1 above)
Comment (akilantech) [reply src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java:63]: agreed

Comment (akilantech) [reply src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java]: Done, Please approve.

Comment (akilantech) [reply src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java]: @olamy   Please help

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java [hunk 2]
```diff
@@ -417,10 +417,16 @@ protected abstract void update( ModifiedPomXMLEventReader pom )
      * @return <code>true</code> if the update should be applied.
      * @since 1.0-alpha-1
      */
-    protected boolean shouldApplyUpdate( Artifact artifact, String currentVersion, ArtifactVersion updateVersion )
+    protected boolean shouldApplyUpdate( Artifact artifact, String currentVersion, ArtifactVersion updateVersion, Boolean forceUpdate )
```
Comment (olamy) [inline src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:420]: please use `boolean` and keep previous method marking it as deprecated and add this new method

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java [hunk 3]
```diff
@@ -417,10 +417,16 @@ protected abstract void update( ModifiedPomXMLEventReader pom )
...
     {
         getLog().debug( "Proposal is to update from " + currentVersion + " to " + updateVersion );
 
+        if ( forceUpdate == Boolean.TRUE )
```
Comment (olamy) [inline src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:424]:
```suggestion
        if ( forceUpdate )
```

File: src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java [hunk 4]
```diff
@@ -54,6 +54,14 @@
     @Parameter( property = "parentVersion", defaultValue = "null" )
     protected String parentVersion = null;
 
+    /**
+     * to update parent version by force when it is RELEASE or LATEST
+     *
+     * @since 2.8
+     */
+    @Parameter( property = "forceUpdate", defaultValue = "False" )
```
Comment (olamy) [inline src/main/java/org/codehaus/mojo/versions/UpdateParentMojo.java:62]:
```suggestion
    @Parameter( property = "forceUpdate", defaultValue = "false" )
```

**Review by olamy — CHANGES_REQUESTED** [review]

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java (hunk 2 above)
Comment (akilantech) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java]: Done as suggested. Please review again.

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java [hunk 5]
```diff
@@ -417,10 +421,61 @@ protected abstract void update( ModifiedPomXMLEventReader pom )
      * @return <code>true</code> if the update should be applied.
      * @since 1.0-alpha-1
      */
+    @Deprecated
     protected boolean shouldApplyUpdate( Artifact artifact, String currentVersion, ArtifactVersion updateVersion )
```
Comment (olamy) [inline src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:425]: the deprecated method should call the new with correct parameter. We should avoid too much duplicated code.

**Review by olamy — CHANGES_REQUESTED** [review]

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java (hunk 5 above)
Comment (akilantech) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:425]: Good point Olamy. Appreciate your patience with me.

Comment (akilantech) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:425]: Pushed the changes. Thanks

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java [hunk 6]
```diff
@@ -417,10 +421,31 @@ protected abstract void update( ModifiedPomXMLEventReader pom )
...
+    }
+
+    /**
+     * Returns <code>true</code> if the update should be applied.
+     *
+     * @param artifact The artifact.
+     * @param currentVersion The current version of the artifact.
+     * @param updateVersion The proposed new version of the artifact.
+     * @return <code>true</code> if the update should be applied to the pom.
+     * @since 1.0-alpha-1
```
Comment (olamy) [inline src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:437]: not sure about the since :) 
```suggestion
     * @since 2.9
```

**Review by olamy — CHANGES_REQUESTED** [review]

**Pull request merged**